deployments/helm/k8s-dra-driver/templates/validatingadmissionpolicy.yaml

```
@@ -0,0 +1,31 @@
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: resourceslices-policy-{{ include "k8s-dra-driver.fullname" . }}
+spec:
+  failurePolicy: Fail
+  matchConstraints:
+    resourceRules:
+    - apiGroups:   ["resource.k8s.io"]
+      apiVersions: ["v1beta1"]
+      operations:  ["CREATE", "UPDATE", "DELETE"]
+      resources:   ["resourceslices"]
+  matchConditions:
+  - name: isRestrictedUser
+    expression: >-
+      request.userInfo.username == "system:serviceaccount:{{ include "k8s-dra-driver.namespace" . }}:{{ include "k8s-dra-driver.serviceAccountName" . }}"
+  variables:
+  - name: userNodeName
+    expression: >-
+      request.userInfo.extra[?'authentication.kubernetes.io/node-name'][0].orValue('')
+  - name: objectNodeName
+    expression: >-
+      (request.operation == "DELETE" ? oldObject : object).spec.?nodeName.orValue("")
+  validations:
+  - expression: variables.userNodeName != ""
+    message: >-
+      no node association found for user, this user must run in a pod on a node and ServiceAccountTokenPodNodeInfo must be enabled
+  - expression: variables.userNodeName == variables.objectNodeName
+    messageExpression: >-
+      "this user running on node '"+variables.userNodeName+"' may not modify " +
+      (variables.objectNodeName == "" ?"cluster resourceslices" : "resourceslices on node '"+variables.objectNodeName+"'")
```

@guptaNswati @klueska wouldn't the below suggestion account for the controller case? For resource slices managed by the controller, `nodeName` will not be set so we can skip the `variables.userNodeName == variables.objectNodeName` check.
```suggestion
  - expression: variables.objectNodeName == "" || variables.userNodeName == variables.objectNodeName
    messageExpression: >-
      "this user running on node '"+variables.userNodeName+"' may not modify " +
      "resourceslices on node '"+variables.objectNodeName+"'"
```

---

Yes, there is no NodeName in resourecslice, we only add the pool name and nodeSelector, so this is not a good check. Look at my above comment.

---

https://github.com/NVIDIA/k8s-dra-driver/pull/219#issuecomment-2549641756:
> admission policy for a pool of nodes and not just a single node

This is a lot less important for a controller that runs in the cluster control plane compared to the DRA driver which runs on the node. That's because on a node, there are additional attack vectors from other pods running on the same node.

I think for a controller you don't need a VAP.

---

Sure make sense. But i think its still valuable to restrict the controller behavior especially when its operating at a group level. Adding simple check of `variables.nodeSelectorValue != ""` wont hurt.